### ISSUE TITLE
When rolling on a Roll Table that is in a Compendium, the "Draw with Replacement" option does not function or throw an error 

### ISSUE THREAD

**Issue by Norc** [foundryvtt/foundryvtt#10603]
### What happened?

When a Roll Table is in a Compendium (either in a Locked or Unlocked state, you can roll on it as usual. However, replacement always happens, even if the "Draw with Replacement"  checkbox is unchecked. No error is thrown in the console.

Very low priority issue, but this did actually come up for a user in the wild. This has been reproduced on several systems and does not seem to be system-related at all.

### What ways of accessing Foundry can you encounter this issue in?

- [X] Native App (Electron)
- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Other

### Reproduction Steps

1. Create a new D&D 5E world
2. In the SRD, toggle the Edit Lock for the Tables compendium
3. Without importing the Bag of Beans table, edit it so that Draw with Replacement is off
4. (optional) Turn the Edit Lock back on for the Tables compendium (does not affect results)
5. Roll against the table a few times

This can also be reproduced by a new table that is then dragged and dropped into a new compendium.

**Expected**: Results that are rolled are not "replaced"- their row is visually darkened, and they cannot be rolled a second time until the roll table is reset. At the very least, an error or a message should be displayed to the user indicating that replacement is not possible for Roll Tables that are not imported into the world.
**Actual**:  Roll is performed exactly as if it is without replacement- the row is not darkened and it can be rolled again without resetting the table.

**Example:**
https://github.com/foundryvtt/foundryvtt/assets/64667579/074b953f-880f-4faa-ba53-df71d9eb7eb8

Note: Screencap is from the band of blades system. 

### What core version are you reporting this for?

11.3115 and 12.318, DND5E 3.1.1

### Bug Checklist

- [X] The issue occurs while all Modules are disabled

**Comment by JPMeehan** [comment]
My guess as to the simple cause is that marking the non-replacement requires an update, which is blocked by being in a locked compendium.

**Comment by Norc** [comment]
While I don't disagree, this happens whether or not the compendium is locked. 

In any case, some sort of a message should be displayed to the user.  I'm particularly surprised that there is no error in the console when this happens, even. I'm curious as to why it doesn't.

If this ever ends up getting addressed, the solution might just be to import the table as a first step and roll from the imported version.

**Comment by Norc** [comment]
Testing in V13.339, this behavior has been improved enough to the point that I think this bug can be closed. Note: change was likely made several V13 releases ago.

Now, when you attempt to roll on a Roll Table that is in a Compendium still, if the "Draw with Replacement" option is unchecked, the "Draw Result" button is automatically disabled. This occurs whether or not the Compendium is locked.

This resolves this as a bug... any further improvements on this behavior should be raised as a feature request.
